src/azul/service/catalog_controller.py

```diff
@@ -28,15 +28,12 @@ def list_catalogs(self) -> schema.object(
                 atlas=str,
                 internal=bool,
                 plugins=schema.object(
-                    repository=schema.object(
+                    additional_properties=schema.object(
                         name=str,
-                        sources=schema.array(str),
-                        additional_properties=False
+                        additional_properties=schema.object(
+                            sources=schema.optional(schema.array(str)),
```

Comment:
@hannes-ucsc pycharm throws a warning here... but the OpenAPI definition is formed correctly, with the nested additional property inside the plugin object not being required....


<img width="563" alt="Screen Shot 2021-07-20 at 5 18 06 PM" src="https://user-images.githubusercontent.com/1947534/126410989-619baa15-45e0-46ba-8231-90a83cfd1ea8.png">

Reply:
Confirmed that the schema is valid against the response
https://www.jsonschemavalidator.net/s/czIhpfAa

<img width="1904" alt="Screen Shot 2021-07-21 at 1 19 16 PM" src="https://user-images.githubusercontent.com/1947534/126554467-ef2d6aa5-bff1-482f-b4ae-a29aea2b8b4f.png">

Reply:
What you pasted on the left isn't a schema. The schema starts after `"schema": ` There must be a bug in the validator you are using.

Reply:
Also, the schema is still wrong. In the response `name` and `sources` are siblings. In the schema they aren't.

Reply:
https://www.jsonschemavalidator.net/s/YcolboWU

Yeah there was a bug in the validator, i changed the schema and it passes validation when using original validator which is linked above. 
Also tested it out with another validator just incase.
<img width="1556" alt="Screen Shot 2021-07-21 at 6 35 27 PM" src="https://user-images.githubusercontent.com/1947534/126579352-1df6cbee-3412-40c6-ae1a-663d9d8437ab.png">

Reply:
`sources` should be required.

Reply:
@hannes-ucsc  Do you want a `sources` field added to the metadata plugin as well? Currently it does not have one.

Reply:
Ahh, no. So `sources` can't be required.